src/main/java/it/gov/pagopa/payment/repository/TransactionInProgressRepositoryImpl.java

```diff
@@ -49,4 +62,37 @@ public UpdateResult createIfExists(TransactionInProgress trx, String trxCode) {
             .setOnInsert(Fields.trxCode, trxCode),
         TransactionInProgress.class);
   }
+
+  @Override
+  public TransactionInProgress findAndModify(String trxCode) {
+    return mongoTemplate.findAndModify(
+        Query.query(Criteria.where(Fields.trxCode).is(trxCode)
+            .andOperator(Criteria.where(Fields.authDate).exists(true)
+                    .orOperator(
+                        Criteria.where(Fields.authDate)
+                            .lte(LocalDateTime.now().minusSeconds(trxThrottlingSeconds)),
+                        Criteria.where(Fields.authDate).is(null)
+                    ),
+                Criteria.where(Fields.trxChargeDate)
+                    .gte(LocalDateTime.now().minusMinutes(trxThrottlingMinutes))
+            )), new Update().set(Fields.authDate, LocalDateTime.now()),
+        FindAndModifyOptions.options().returnNew(true), TransactionInProgress.class);
```

Comment:
if null, you should check if exists searching just by trxCode and chargeDate
(use mongoTemplate.exists method)

if it exists, you should throw a ClientNoBodyException with HttpStatus.TOO_MANY_REQUESTS